### DISCUSSION TITLE
Getting a constructors most recent active drivers

### DISCUSSION THREAD

**Discussion by JackThomas** [jolpica/jolpica-f1#101] [Q&A] [answered]
Is there an endpoint or anything I can do to get the current drivers for a constructor?

I've was looking at mapping the /2024/driverconstructors endpoint response as this returns the Driver object with an array of Constructors, the only downside to this is that there isnt anything to specify if the driver is currently active.

Eg. Williams would give 3 drivers (Albon, Colapinto and Sargeant) but theres nothing to say who is currently driving.

**Comment by jolpica** [comment, marked as answer]
You can specifiy the round number to get drivers from, for example:
https://api.jolpi.ca/ergast/f1/2024/20/constructors/williams/drivers/

would return:
```json
{
    "MRData": {
        "xmlns": "",
        "series": "f1",
        "url": "http://api.jolpi.ca/ergast/f1/2024/20/constructors/williams/drivers/",
        "limit": "30",
        "offset": "0",
        "total": "2",
        "DriverTable": {
            "season": "2024",
            "round": "20",
            "constructorId": "williams",
            "Drivers": [
                {
                    "driverId": "albon",
                    "permanentNumber": "23",
                    "code": "ALB",
                    "url": "http://en.wikipedia.org/wiki/Alexander_Albon",
                    "givenName": "Alexander",
                    "familyName": "Albon",
                    "dateOfBirth": "1996-03-23",
                    "nationality": "Thai"
                },
                {
                    "driverId": "colapinto",
                    "permanentNumber": "43",
                    "code": "COL",
                    "url": "http://en.wikipedia.org/wiki/Franco_Colapinto",
                    "givenName": "Franco",
                    "familyName": "Colapinto",
                    "dateOfBirth": "2003-05-27",
                    "nationality": "Argentinian"
                }
            ]
        }
    }
}
```

You can also remove the constructor filter to get a list of all currently active drivers in a given round

**Reply by JackThomas** [reply]
Great, thank you!

**Reply by JackThomas** [reply]
I've just been playing around with this. When I attempt to get the Drivers list for each Constructor, I get a "Request was throttled." message, is there any way around that? 
I attempted https://api.jolpi.ca/ergast/f1/2024/20/constructors/drivers/ (without the constructorId, but that didn't resolve to a route.
Would having an all constructorId work? https://api.jolpi.ca/ergast/f1/2024/20/constructors/all/drivers/

**Reply by jolpica** [reply]
If you remove the filter entirely it should work for you
https://api.jolpi.ca/ergast/f1/2024/20/drivers/

You are likely getting Request was throttled as you are making too many requests (either more than 4 per second or 500 per hour). Using the above endpoint instead should reduce the requests, and therefore limit the throttling.

The API scheme is a bit annoying to work with, as its adapted from ergast, hopefully next year we'll have a scheme thats a bit easier to work with.

**Reply by jolpica** [reply]
Also, we've recently added some documentation for the API to the repository. Please let us know if theres anything we can add to improve it
https://github.com/jolpica/jolpica-f1/blob/main/docs/endpoints/drivers.md

**Reply by JackThomas** [reply]
> If you remove the filter entirely it should work for you https://api.jolpi.ca/ergast/f1/2024/20/drivers/

This is basically what I'm after, but there are no references to the constructor to sort the drivers.
I think I might be able to do it by using the https://api.jolpi.ca/ergast/f1/2024/20/results/ and mapping them

**Reply by jolpica** [reply]
Ah yes, I see what you mean.
Yeah I think the results endpoint would likely be best for what you need in that case then unfortunately.

Also, if you weren't already aware you can use [/current/last/](https://api.jolpi.ca/ergast/f1/current/last/results/) to automatically get the latest race without having to specify a round. One downside of this is that blank data may be returned on Mondays if there is a bit of delay in data ingest (hopefully most weeks there shouldn't be any!)

**Reply by JackThomas** [reply]
Oh, that should be perfect. Thanks for taking the time to respond!
